src/shared/Eis.cpp

```diff
@@ -0,0 +1,286 @@
+    eis_seat*   seat;
+    eis_device* device;
+
+    switch (eis_event_get_type(e)) {
+        case EIS_EVENT_CLIENT_CONNECT:
+            client = eis_event_get_client(e);
+            Debug::log(LOG, "[EIS] {} client connected: {}", eis_client_is_sender(client) ? "sender" : "receiver", eis_client_get_name(client));
+
+            if (eis_client_is_sender(client)) {
+                Debug::log(WARN, "[EIS] Unexpected sender client {} connected to input capture session", eis_client_get_name(client));
+                eis_client_disconnect(client);
+                return 0;
+            }
+
+            if (this->client.handle != nullptr) {
+                Debug::log(WARN, "[EIS] Unexpected additional client {} connected to input capture session", eis_client_get_name(client));
+                eis_client_disconnect(client);
+                return 0;
+            }
+
+            this->client.handle = client;
+
+            eis_client_connect(client);
+            Debug::log(LOG, "[EIS] creating new default seat");
+            seat = eis_client_new_seat(client, "default");
+
+            eis_seat_configure_capability(seat, EIS_DEVICE_CAP_POINTER);
+            eis_seat_configure_capability(seat, EIS_DEVICE_CAP_BUTTON);
+            eis_seat_configure_capability(seat, EIS_DEVICE_CAP_SCROLL);
+            eis_seat_configure_capability(seat, EIS_DEVICE_CAP_KEYBOARD);
+            eis_seat_add(seat);
+            this->client.seat = seat;
+            break;
+        case EIS_EVENT_CLIENT_DISCONNECT:
+            client = eis_event_get_client(e);
+            Debug::log(LOG, "[EIS] {} disconnected", eis_client_get_name(client));
+            eis_client_disconnect(client);
+
+            eis_seat_unref(this->client.seat);
+            clearPointer();
+            clearKeyboard();
+            this->client.handle = NULL;
+            break;
+        case EIS_EVENT_SEAT_BIND:
+            Debug::log(LOG, "[EIS] Binding seats...");
+
+            if (eis_event_seat_has_capability(e, EIS_DEVICE_CAP_POINTER) && eis_event_seat_has_capability(e, EIS_DEVICE_CAP_BUTTON) &&
+                eis_event_seat_has_capability(e, EIS_DEVICE_CAP_SCROLL))
+                ensurePointer(e);
+            else
+                clearPointer();
+
+            if (eis_event_seat_has_capability(e, EIS_DEVICE_CAP_KEYBOARD))
+                ensureKeyboard(e);
+            else
+                clearKeyboard();
+            break;
+        case EIS_EVENT_DEVICE_CLOSED:
+            device = eis_event_get_device(e);
+            if (device == this->client.pointer) {
+                clearPointer();
+            } else if (device == this->client.keyboard) {
+                Debug::log(LOG, "[EIS] Clearing keyboard");
+                clearKeyboard();
+            } else {
+                Debug::log(WARN, "[EIS] Unknown device to close");
+            }
+            break;
+        case EIS_EVENT_FRAME: Debug::log(LOG, "[EIS] Got event EIS_EVENT_FRAME"); break;
```

Comment:
shouldn't these do something more than just logging? Also, shouldn't these be dropped to TRACE? Kinda verbose.

Reply:
I forgot to remove them.